packages/@aws-cdk/aws-appconfig-alpha/lib/environment.ts

```diff
@@ -274,7 +274,21 @@ export class Environment extends EnvironmentBase {
     this.application.addExistingEnvironment(this);
   }
 
-  private createAlarmRole(alarmArn: string, index: number): iam.IRole {
+  private createAlarmRole(monitor: Monitor, index: number): iam.IRole {
+    const roleHash = monitor.isCompositeAlarm ? 5 : index;
+    const logicalId = `Role${roleHash}`;
```

Comment:
```suggestion
    const logicalId = monitor.isCompositeAlarm ? 'RoleCompositeAlarm' : `Role${index}`;
```
Will not depend on `monitors` length constraints.